runtime/compiler/codegen/J9CodeGenerator.cpp

```diff
@@ -156,6 +154,172 @@ static TR::Node *lowerCASValues(
    }
 
 
+/**
+ * @brief Add checks to skip (fast-path) acmpHelper call
+ *
+ * @details
+ *
+ * This transformation adds checks for the cases where the acmp can be performed
+ * without calling the VM helper. The trasformed Trees represen the following operation:
+ *
+ * 1. If the address of lhs and rhs are the same, produce an eq (true) result
+ *    and skip the call (note the two objects must be the same regardless of
+ *    whether they are value types are reference types)
+ * 2. Otherwise, do VM helper call
+ *
+ * The transformation looks as follows:
+ *
+ *  +----------------------+
+ *  |ttprev                |
+ *  |treetop               |
+ *  |  icall acmpHelper    |
+ *  |    aload lhs         |
+ *  |    aload rhs         |
+ *  |ificmpeq --> ...      |
+ *  |  ==> icall           |
+ *  |  iconst 0            |
+ *  |BBEnd                 |
+ *  +----------------------+
+ *
+ *  ...becomes...
+ *
+ *  +----------------------+
+ *  |ttprev                |
+ *  |iRegStore x           |
+ *  |  iconst 1            |
+ *  |ifacmpeq  -->---------*---------+
+ *  |  aload lhs           |         |
+ *  |  aload rhs           |         |
+ *  |BBEnd                 |         |
+ *  +----------------------+         |
+ *  |BBStart (extension)   |         |
+ *  |iRegStore x           |         |
+ *  |  icall acmpHelper    |         |
+ *  |    aload lhs         |         |
+ *  |    aload rhs         |         |
+ *  |BBEnd                 |         |
+ *  +----------------------+         |
+ *        |                          |
+ *        +--------------------------+
+ *        |
+ *        v
+ *  +-----------------+
+ *  |BBStart
+ *  |ificmpeq --> ... |
+ *  |  iRegLoad x     |
+ *  |  iconst 0       |
+ *  |BBEnd            |
+ *  +-----------------+
+ *
+ */
+void
+J9::CodeGenerator::fastpathAcmpHelper(TR::Node *node, TR::TreeTop *tt, const bool trace)
+   {
+   TR::Compilation* comp = self()->comp();
+   TR::CFG* cfg = comp->getFlowGraph();
+   cfg->setStructure(0);
```

Comment:
I think this should use invalidateStructure rather than setStructure(0). That being the new API to invalidate structure.

Reply:
Fixed in force push.